elements/howto-tree/howto-tree.js

```diff
@@ -0,0 +1,497 @@
+     * It's important to note that there are no side effects taking place in
+     * the property setter. For example, the setter does not set
+     * `aria-selected`.
+     * Instead, that work happens in the `attributeChangedCallback`.
+     * As a general rule, make property setters very dumb, and if setting a
+     * property or attribute should cause a side effect (like setting a
+     * corresponding ARIA attribute) do that work in the
+     * `attributeChangedCallback`. This will avoid having to manage complex
+     * attribute/property reentrancy scenarios.
+     */
+    set selected(value) {
+      const isSelected = Boolean(value);
+      if (isSelected)
+        this.setAttribute('selected', '');
+      else
+        this.removeAttribute('selected');
+    }
+
+    get selected() {
+      return this.hasAttribute('selected');
+    }
+
+    attributeChangedCallback(name, value) {
+      if (name === 'selected')
+        this.setAttribute('aria-selected', this.selected);
+    }
+  }
+
+  /**
+   * Define a custom element, `<howto-tree-item>`, and associate it with the
+   * `HowToTreeItem` class.
+   */
+  window.customElements.define('howto-tree-item', HowToTreeItem);
+
+  /**
+   * `HowToTreeItemGroup` is a simple container that holds `HowToTreeItem`
+   * children and can be expanded or collapsed.
+   * Because a `HowToTreeItemGroup` is also a treeItem, and can be selected,
+   * it inherits from the `HowToTreeItem` element.
+   */
+  class HowToTreeItemGroup extends HowToTreeItem {
+    /**
+     * Append an `expanded` attribute, to the list of `observedAttributes`.
+     * This getter also demonstrates a pattern for guarding against a situation
+     * where the parent may not have defined any `observedAttributes`, in
+     * which case an empty array is used.
+     */
+    static get observedAttributes() {
+      return (super.observedAttributes || []).concat(['expanded']);
+    }
+
+    connectedCallback() {
+      super.connectedCallback();
+      // An expandable treeItem must have an explicit `aria-expanded` value.
+      // This handles the case where the element starts without an
+      // `[expanded]` attribute. In that scenario, the element would be set
+      // to `aria-expanded=false`.
+      this.setAttribute('aria-expanded', this.expanded);
+
+      // This first child should be a `<label>` element. Custom Elements are
+      // not currently supported by the `<label>` element, but hopefully
+      // they will be in the future. In the meantime, set the `aria-label`
+      // for the `HowToTreeItem`, equal to the `<label>` text.
+      // Without this labeling, the element's name will be computed based on
+      // its text content plus the text content of all of its children, making
+      // it so verbose as to be unusable.
+      if (!this.hasAttribute('aria-label')) {
+        let label = this.querySelector('label');
+        if (!label) {
+          console.error(`The first child of a <howto-tree-item> that ` +
```

Comment:
I have been throwing errors in these kinda situations. Logic being that the element cannot function if this requirement wasn’t met. What do we wanna settle for?